src/test/scala/com/advancedtelematic/campaigner/data/Generators.scala

```diff
@@ -38,7 +38,7 @@ object Generators {
   val genCreateCampaign: Gen[CreateCampaign] = for {
     n   <- arbitrary[String]
     update <- arbitrary[UpdateId]
-    gs  <- arbitrary[Set[GroupId]]
+    gs  <- arbitrary[Set[GroupId]].retryUntil(_.nonEmpty)
```

Comment:
You should change the generator in Generators instead, look for Gen.listOf methods.

Reply:
I changed this generator now.